tst/testrandom.g

```diff
@@ -49,45 +72,79 @@ randomTest := function(collection, method, checkin...)
     fi;
 end;;
 
+
 # A special test for collections of size 1
-randomTestForSizeOneCollection := function(collection, method)
-    local i, val, localgen, intlist1, intlist2;
-    if Size(collection) <> 1 then
-        Print("randomTestForSizeOneCollection is only for collections of size 1");
-        return;
-    fi;
+# Here we can't check different seeds produce different answers
+# We do check that the random source is not used, for efficency.
+randomTestForSizeOneCollectionInner := function(filter, global_rs, global_randfunc, randfunc, collection, checkin)
+    local i, val, local_rs, intlist1, intlist2;
 
     val := Representative(collection);
 
-    Init(GlobalMersenneTwister, 6);
-    intlist1 := List([1..1000], x -> Random([1..10]));
+    Init(global_rs, 6);
+    intlist1 := List([1..10], x -> global_randfunc([1..1000]));
 
-    for i in [1..1000] do
-        if method(collection) <> val then
+    for i in [1..100] do
+        if global_randfunc(collection) <> val then
             Print("Random returned something outside collection :", collection, ":", val);
         fi;
     od;
 
-    for i in [1..1000] do
-        if method(GlobalMersenneTwister, collection) <> val then
+    for i in [1..100] do
+        if randfunc(global_rs, collection) <> val then
             Print("Random returned something outside collection :", collection, ":", val);
         fi;
     od;
 
-    localgen := RandomSource(IsMersenneTwister, 6);
+    local_rs := RandomSource(filter, 6);
 
-    Init(GlobalMersenneTwister, 6);
-    for i in [1..1000] do
-        if method(localgen, collection) <> val then
+    Init(global_rs, 6);
+    for i in [1..100] do
+        if randfunc(local_rs, collection) <> val then
             Print("Random returned something outside collection :", collection, ":", val);
         fi;
     od;
 
-    # The previous loop should not have affected GlobalMersenneTwister,
+    # The previous loop should not have affected global_rs,
     # so this should be the same as intlist1
-    intlist2 := List([1..1000], x -> Random([1..10]));
+    intlist2 := List([1..10], x -> global_randfunc([1..1000]));
```

Comment:
Why not just query the state before and after, and compare it? Make an immutable copy to make sure it's not been modified?

Reply:
I quite like testing the actual sequence. However, it might be nice to also test the states are equal as well!